Client/Frontend/Home/ActivityStreamTopSitesCell.swift

```diff
@@ -123,10 +120,14 @@ class TopSiteItemCell: UICollectionViewCell {
         contentView.backgroundColor = UIColor.clear
         imageView.image = nil
         imageView.backgroundColor = UIColor.clear
+        faviconBG.backgroundColor = UIColor.clear
+        imageView.sd_cancelCurrentImageLoad()
```

Comment:
Sadly no. Because the color calculations also happen in a different thread sometimes the cell could be reused after the imageLoad but before the color operation calls the completionBlock.